Controller/ResetPasswordController.php

```diff
@@ -0,0 +1,196 @@
+     * @param UserManager $userManager
+     * @param TokenGeneratorInterface $tokenGenerator
+     * @param MailerInterface $mailer
+     * @param RouterInterface $router
+     * @param Security $security
+     * @param EncoderFactoryInterface $encoderFactory
+     * @param array $fromEmail
+     * @param string $userClass
+     */
+    public function __construct(UserManager $userManager, TokenGeneratorInterface $tokenGenerator, MailerInterface $mailer, RouterInterface $router, Security $security, EncoderFactoryInterface $encoderFactory, array $fromEmail, string $userClass)
+    {
+        $this->tokenGenerator = $tokenGenerator;
+        $this->userManager = $userManager;
+        $this->mailer = $mailer;
+        $this->router = $router;
+        $this->security = $security;
+        $this->encoderFactory = $encoderFactory;
+        $this->fromEmail = $fromEmail;
+        $this->userClass = $userClass;
+    }
+
+    /**
+     * @Template()
+     */
+    public function request(): array
+    {
+        return [];
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     * @throws TransportExceptionInterface
+     */
+    public function sendmail(Request $request): Response
+    {
+        $username = $request->request->get('username');
+
+        $user = $this->userManager->findUser($username);
+
+        if (null !== $user) {
+            if (null === $user->getConfirmationToken()) {
+                $user->setConfirmationToken($this->tokenGenerator->generateToken());
+            }
+
+            $user->setPasswordRequestedAt(new DateTime());
+
+            $this->getDoctrine()->getManager()->flush();
+
+            $email = (new TemplatedEmail())
+                ->from(new Address($this->fromEmail['address'], $this->fromEmail['sender_name']))
+                ->to($user->getEmail())
+                ->subject('Wachtwoord resetten')
+                ->htmlTemplate('@WebstackUser/email/reset-password/confirm.html.twig')
+                ->context([
+                    'user' => $user,
+                    'confirmationUrl' => $this->router->generate('webstack_user_reset_password_reset', [
+                        'token' => $user->getConfirmationToken()
+                    ], UrlGeneratorInterface::ABSOLUTE_URL)
+                ]);
+
+            $this->mailer->send($email);
+        }
+
+        return $this->redirectToRoute('webstack_user_reset_password_check_email', [
+            'username' => $username
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @return array|RedirectResponse
+     */
+    public function checkEmail(Request $request): Response
+    {
+        $username = $request->query->get('username');
+
+        if (empty($username)) {
+            return new RedirectResponse($this->generateUrl('webstack_user_reset_password_index'));
+        }
+
+        return $this->render('@WebstackUser/reset_password/check_email.html.twig', [
+            'email' => $username
```

Comment:
Graag hier consistent zijn, of email of username gebruiken als variabele naam en waarde